src/main/java/org/mastodon/mamut/clustering/util/Classification.java

```diff
@@ -84,6 +92,9 @@ public Classification( final List< Pair< Set< T >, Cluster > > classifiedObjects
 		}
 		this.rootCluster = rootCluster;
 		this.cutoff = cutoff;
+
+		double[] nonZeroValues = Stream.of( distances ).flatMapToDouble( DoubleStream::of ).filter( value -> value != 0 ).toArray();
+		this.median = nonZeroValues.length == 0 ? 0 : Util.median( nonZeroValues );
```

Comment:
I would suggest to compute the median in the `getClassificationByThreshold(...)` method. This constructor would then simply have a parameter `double medianDistance`.

What's the reasoning behind this suggestion:
1. The important computations would be gathered in one method `getClassificationByThreshold(...)`
2. The class `Classifiaction` would do no computation only store data. That way it's easier to understand. And it would also match you javadoc for the `Classification` class, which states: "A class that encapsulates the result of a clustering algorithm"

Reply:
Thanks for the advice. I also thought about this before, but hesitated to do this as suggested, since the Classification constructor existed 3 times before due to the handling of special cases in `getClassificationByClassCount()`. I omitted handling these special cases there, which allowed to do the median computation at only one location in the production code.